src/ui/control/geolocate_control.js

```diff
@@ -120,6 +127,7 @@ class GeolocateControl extends Evented {
         this._map = map;
         this._container = DOM.create('div', `mapboxgl-ctrl mapboxgl-ctrl-group`);
         checkGeolocationSupport(this._setupUI);
+        this._map.on('zoom', this._onZoom);
```

Comment:
When the map is heavily pitched, the scale of the dot will change as you pan vertically, so it might not be enough to subscribe to `zoom` — should we do `move` instead?